pkg/cloudprovider/azure/common.go

```diff
@@ -31,6 +31,7 @@ const (
 	subscriptionIDEnvVar = "AZURE_SUBSCRIPTION_ID"
 	clientIDEnvVar       = "AZURE_CLIENT_ID"
 	clientSecretEnvVar   = "AZURE_CLIENT_SECRET"
+	cloudNameEnvVar      = "AZURE_CLOUD_NAME"
```

Comment:
I think my preference would be to have this as a config entry within the `BackupStorageLocation`, so more like `resourceGroupConfigKey`, though I guess the downside of that is that it would need to be specified for both the `BackupStorageLocation` and `VolumeSnapshotLocation`. cc @nrb @carlisia @prydonius for thoughts on specifying this as an env var within the secret vs. a config item.

Reply:
I also feel like this would be better placed in the `BackupStorageLocation` and `VolumeSnapshotLocation`, but as you said an env var does help avoid having to specify it in two places. That said, `velero install` makes that a little easier.

Reply:
hi @skriss @prydonius thanks for the review.
I would say `AZURE_CLOUD_NAME` should be the essential field as other fields, e.g. `AZURE_SUBSCRIPTION_ID`, user should specify this field in the beginning since same azure subscription id does not work in different azure clouds, e.g. `AzureGermanCloud`, `AzureChinaCloud`, you could see this [cloud-provider-config](https://github.com/kubernetes/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#cloud-provider-config), `cloud` is an essential field.

While for comparability reason, if customer don't set `AZURE_CLOUD_NAME` env, we use default `AzurePublicCloud`.

`AZURE_SUBSCRIPTION_ID`, `AZURE_CLIENT_ID`, `AZURE_CLOUD_NAME` should be in one config file since different azure clouds has different SubsiprtionID, clientID, user must be aware of that. e.g. One SubsiprtionID is valid in `AzurePublicCloud`, while it won't be valid in `AzureGermanCloud`, SubsiprtionID should be bound to CloudName.

Reply:
OK - given that a subscription is limited to a single Azure cloud, and therefore every BSL/VSL in a given Velero install would have the same value, I'm fine including this in the secret.